tanglegram/tangle.py

```diff
@@ -754,62 +651,28 @@ def leaf_order(link, labels=None, as_dict=True):
             return leafs_ix
 
 
-def entanglement(lindex1, lindex2, L=1.5):
-    """Calculage average displacement of leafs in dendogram 1 and 2.
-
-    Entanglement is a measure between 1 (full entanglement) and 0 (no
-    entanglement). Ignores leafs that aren't present in both dendrograms.
-
-    Parameters
-    ----------
-    lindex1,lindex2 :       dict
-                            Dictionaries mapping the labels of two dendrograms
-                            to their indices.
-    L :                     any positive number
-                            Distance norm to use for measuring the distance
-                            between the two trees. Can be any positive number,
-                            often one will want to use 0, 1, 1.5 or 2:
-                            ``sum(abs(x-y)^L)``.
-
+def entanglement(link1, link2):
+    """ Entanglement is measured by giving the left tree’s labels the values of 1 till tree size,
+    matching these numbers with the right tree, and then dividing the sum of the square difference
+    between elements of these two vectors (sum(abs(x-y)^2)) by that number of the "worst case" entanglement
+    (e.g: when the right tree is the complete reverse of the left tree).
+    Skips leafs that aren't present in both dendrograms.
     """
-    assert isinstance(lindex1, dict)
-    assert isinstance(lindex2, dict)
+    lindex1 = leaves_list(link1)
+    lindex2 = leaves_list(link2)
 
     exist_in_both = list(set(lindex1) & set(lindex2))
+    ix = np.arange(max(len(lindex1), len(lindex2)))
 
     if not exist_in_both:
```

Comment:
If you are not using labels but just indices, then there is no point in checking if they exist in both. Or am I missing something?

Reply:
The "leaves_list" function returns the list of leaves' indices. So we only work with indices. In doing so we have to assume that the relationship between indices and labels is one-to-one and identical in both trees.

Reply:
That's fair but I want/need to cater for scenarios where that's not the case.

Reply:
Totally agree. We should use labels instead of indices.